### PR TITLE
Update character-set-and-collation.md

### PR DESCRIPTION
**Pull request by ireneontheway**
### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/3256
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

File: character-set-and-collation.md [hunk 1]
```diff
@@ -6,7 +6,15 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
...
 # Character Set and Collation
 
-A character set is a set of symbols and encodings. A collation is a set of rules for comparing characters in a character set.
+This document introduces the character set and collation supported by TiDB.
+
+## Concepts
+
+A character set is a set of symbols and encodings. 
+
+A collation is a set of rules for comparing characters in a character set.
```
Comment (nullnotnil) [inline character-set-and-collation.md:15]: May I suggest the following as an intro:

A character set is a set of symbols and encodings. The default character set in TiDB is utf8mb4, which matches the default in MySQL 8.0 and above. In TiDB `utf8` and `utf8mb4` behave identically, and `utf8` is not restricted to a maximum of 3 bytes as in MySQL. UTF-8 encoding [accounts for between 83% - 100% of webpages](https://en.wikipedia.org/wiki/UTF-8), depending on the language and country.

A collation is a set of rules for comparing characters in a character set, and the sorting order of characters. For example in a binary collation `A` and `a` do not compare as equal:

{{< copyable "sql" >}}

```sql
SET NAMES utf8mb4 COLLATE utf8mb4_bin;
SELECT 'A' = 'a';
SET NAMES utf8mb4 COLLATE utf8mb4_general_ci;
SELECT 'A' = 'a';
```

```sql
mysql> SELECT 'A' = 'a';
+-----------+
| 'A' = 'a' |
+-----------+
|         0 |
+-----------+
1 row in set (0.00 sec)

mysql> SET NAMES utf8mb4 COLLATE utf8mb4_general_ci;
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT 'A' = 'a';
+-----------+
| 'A' = 'a' |
+-----------+
|         1 |
+-----------+
1 row in set (0.00 sec)
```

TiDB defaults to using a binary collation. This differs from MySQL, which uses a case-insensitive collation by default.

Comment (TomShawn) [reply character-set-and-collation.md:15]: @wjhuang2016 PTAL

Comment (wjhuang2016) [reply character-set-and-collation.md:15]: In TiDB utf8 and utf8mb4 behave identically, and utf8 is not restricted to a maximum of 3 bytes as in MySQL.

In fact, TiDB checks if the bytes character is greater than 3 unless `check-mb4-value-in-utf8` is disabled.

Comment (yikeke) [reply character-set-and-collation.md:15]: @nullnotnil Do you need to modify your suggested intro according to https://github.com/pingcap/docs/pull/3402#discussion_r459830078?

File: character-set-and-collation.md [hunk 2]
```diff
@@ -255,12 +279,13 @@ You can use the following statement to set the character set and collation that
     ```sql
     SET character_set_client = charset_name;
     SET character_set_results = charset_name;
+    SET charset_connection = @@charset_database;
     SET collation_connection = @@collation_database;
     ```
 
 ## Optimization levels of character sets and collations
```
Comment (yikeke) [inline character-set-and-collation.md:286]:
```suggestion
## Selection priorities of character sets and collations
```

File: character-set-and-collation.md [hunk 3]
```diff
@@ -345,13 +370,13 @@ If an expression involves multiple clauses of different collations, you need to
 
 + The coercibility value of the explicit `COLLATE` clause is `0`.
 + If the collations of two strings are incompatible, the coercibility value of the concatenation of two strings with different collations is `1`. Currently, all implemented collations are compatible with each other.
-+ The column's collation has a coercibility value of `2`.
++ The collation of the column,`CAST()`, `CONVERT()` or `BINARY()` has a coercibility value of `2`.
```
Comment (yikeke) [inline character-set-and-collation.md:373]:
```suggestion
+ The collation of the column, `CAST()`, `CONVERT()`, or `BINARY()` has a coercibility value of `2`.
```

File: character-set-and-collation.md (hunk 1 above)
Comment (ireneontheway) [reply character-set-and-collation.md]: I've updated it.

**Comment by yikeke** [comment]
PTAL again @nullnotnil @wjhuang2016

File: character-set-and-collation.md (hunk 1 above)
Comment (nullnotnil) [reply character-set-and-collation.md]: > @nullnotnil Do you need to modify your suggested intro according to [#3402 (comment)](https://github.com/pingcap/docs/pull/3402#discussion_r459830078)?

Yes, I verified and my earlier comment was incorrect. I have updated it to describe the check-mb4 option. I will add a section on utf8mb4 vs utf8.

File: character-set-and-collation.md [hunk 4]
```diff
@@ -51,11 +107,11 @@ SHOW COLLATION WHERE Charset = 'utf8mb4';
 2 rows in set (0.00 sec)
 ```
```
Comment (nullnotnil) [inline character-set-and-collation.md:109]: ## UTF-8 in MySQL

In MySQL, the character set `utf8` is limited to a maximum of three bytes. This is sufficient to store characters in the Basic Multilingual Plane (BMP), but not enough to store characters such as emojis. For this, it is recommended to use the character set `utf8mb4` instead.

By default, TiDB provides the same 3-byte limit on `utf8` to ensure that data created in TiDB can still safely be restored in MySQL. This checking can be disabled by changing the value of `check-mb4-value-in-utf8` to `FALSE` in your TiDB configuration file.

The following demonstrates the default behavior when inserting a 4-byte emoji character into a table. The `INSERT` statement fails for the `utf8` character set, but succeeds for the `ut8mb4`:

```sql
mysql> CREATE TABLE utf8_test (
    ->  c char(1) NOT NULL
    -> ) CHARACTER SET utf8;
Query OK, 0 rows affected (0.09 sec)

mysql> CREATE TABLE utf8m4_test (
    ->  c char(1) NOT NULL
    -> ) CHARACTER SET utf8mb4;
Query OK, 0 rows affected (0.09 sec)

mysql> INSERT INTO utf8_test VALUES ('😉');
ERROR 1366 (HY000): incorrect utf8 value f09f9889(😉) for column c
mysql> INSERT INTO utf8m4_test VALUES ('😉');
Query OK, 1 row affected (0.02 sec)

mysql> SELECT char_length(c), length(c), c FROM utf8_test;
Empty set (0.01 sec)

mysql> SELECT char_length(c), length(c), c FROM utf8m4_test;
+----------------+-----------+------+
| char_length(c) | length(c) | c    |
+----------------+-----------+------+
|              1 |         4 | 😉     |
+----------------+-----------+------+
1 row in set (0.00 sec)
```

Comment (yikeke) [reply character-set-and-collation.md:109]: May I suggest changing the header to ``` `utf8` and `ut8mb4` in TiDB ``` to focus our topic on TiDB?  @nullnotnil

File: character-set-and-collation.md [hunk 5]
```diff
@@ -6,7 +6,47 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
 
 # Character Set and Collation
 
-A character set is a set of symbols and encodings. A collation is a set of rules for comparing characters in a character set.
+This document introduces the character set and collation supported by TiDB.
+
+## Concepts
+
+A character set is a set of symbols and encodings. The default character set in TiDB is utf8mb4, which matches the default in MySQL 8.0 and above. In TiDB `utf8` and `utf8mb4` behave identically, and `utf8` is not restricted to a maximum of 3 bytes as in MySQL. UTF-8 encoding [accounts for between 83% - 100% of webpages](https://en.wikipedia.org/wiki/UTF-8), depending on the language and country.
```
Comment (yikeke) [inline character-set-and-collation.md:13]:
```suggestion
A character set is a set of symbols and encodings. The default character set in TiDB is utf8mb4, which matches the default in MySQL 8.0 and above.
```
Let's delete them after your new section is added? @nullnotnil

File: character-set-and-collation.md (hunk 4 above)
Comment (nullnotnil) [reply character-set-and-collation.md:109]: Works for me

Comment (yikeke) [reply character-set-and-collation.md:109]: Please apply the suggestion~ @ireneontheway

Comment (ireneontheway) [reply character-set-and-collation.md:109]: Done.

File: character-set-and-collation.md [hunk 6]
```diff
@@ -6,7 +6,47 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
 
 # Character Set and Collation
 
-A character set is a set of symbols and encodings. A collation is a set of rules for comparing characters in a character set.
+This document introduces the character set and collation supported by TiDB.
```
Comment (yikeke) [inline character-set-and-collation.md:9]:
```suggestion
This document introduces the character sets and collations supported by TiDB.
```

**Comment by ireneontheway** [comment]
PTAL again @nullnotnil @wjhuang2016

File: character-set-and-collation.md [hunk 7]
```diff
@@ -6,7 +6,47 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
...
+| 'A' = 'a' |
++-----------+
+|         1 |
++-----------+
+1 row in set (0.00 sec)
+```
+
+TiDB defaults to using a binary collation. This differs from MySQL, which uses a case-insensitive collation by default.
+
+## TiDB character set and collation
```
Comment (yikeke) [inline character-set-and-collation.md:49]:
```suggestion
## Character sets and collations supported by TiDB
```

**Comment by nullnotnil** [comment]
LGTM

File: character-set-and-collation.md [hunk 8]
```diff
@@ -6,7 +6,47 @@ aliases: ['/docs/dev/character-set-and-collation/','/docs/dev/reference/sql/char
...
+
+```sql
+SET NAMES utf8mb4 COLLATE utf8mb4_bin;
+SELECT 'A' = 'a';
+SET NAMES utf8mb4 COLLATE utf8mb4_general_ci;
+SELECT 'A' = 'a';
+```
+
+```sql
+mysql> SELECT 'A' = 'a';
```
Comment (wjhuang2016) [inline character-set-and-collation.md:27]: Please remove the leading `mysql>`

Comment (ireneontheway) [reply character-set-and-collation.md]: Done

Comment (yikeke) [reply character-set-and-collation.md]: Why removed it? We use this leading `mysql>` a lot in our documents.

Comment (wjhuang2016) [reply character-set-and-collation.md:27]: According to @TomShawn, if we use this leading `mysql>`. Then if users copy this SQL, they need to trim the leading `mysql>` again.

Comment (yikeke) [reply character-set-and-collation.md:27]: [lines 17-24](https://github.com/pingcap/docs/pull/3402/files#diff-c8c1ce90ea28120a4eb9bd98c70a9213R17-R24) is the SQL code for copying and the leading `mysql>` needs to be removed. 

lines 26-45 are like examples of the execution result, so no need to delete the leading `mysql>`. @wjhuang2016

**Pull request merged**